promql/functions.go

```diff
@@ -404,11 +404,22 @@ func funcDoubleExponentialSmoothing(vals []parser.Value, args parser.Expressions
 	return append(enh.Out, Sample{F: s1}), nil
 }
 
+func filterFloats(vals []parser.Value) Vector {
```

Comment:
This is very expensive, as you have to copy all the values into a new slice. It could be a lot.

As the original code sorts the vector in place, it should be fine to do also the filtering in place.

So how about going through the vector and remove histograms in place with something like https://go.dev/wiki/SliceTricks#filtering-without-allocating ?

Reply:
I should clarify that not the copying is expensive but the allocation. With the proposed change, we still copy, but just within the existing backing array, so no allocation.

Reply:
filtering without allocating will be less expensive and will be optimal. I'll apply the changes.

Reply:
Applied the suggested changes.